### PR TITLE
[MM-36050] Change enterprise license check for installing multiple Jira instances to allow for E20, Professional, and Enterprise

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

The license check is now using the same logic as the [incident collaboration plugin](https://github.com/mattermost/mattermost-plugin-incident-collaboration/blob/3effb0648a9e8ceada969520fe127ac11f34af88/server/app/permissions.go#L231), which is using logic from mattermost-plugin-api that [checks the SKU](https://github1s.com/mattermost/mattermost-plugin-api/blob/HEAD/license.go#L45) of the license.

Docs have been updated, as well as the `enterprise/LICENSE` file, based off of https://github.com/mattermost/mattermost-plugin-incident-collaboration/pull/540/files

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-36050

### CODE REVIEW

**Review by catalintomai — APPROVED** [review]

File: readme.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 This plugin supports a two-way integration between Mattermost and Jira. Jira Core and Jira Software products, for Server, Data Center, and Cloud platforms are supported. It has been tested with versions 7 and 8.
 
-From v3.0 of this plugin, support for multiple Jira instances is offered for Mattermost Enterprise Edition E20.
+From v3.0 of this plugin, one Mattermost instance can use multiple Jira instances at once.
```
Comment (aaronrothschild) [inline readme.md:5]: @mickmister is it v 3.01 or whatever we are releasing this fix as?

**Review by aaronrothschild — APPROVED** [review]

**Review by jfrerich — APPROVED** [review]
When was it decided the enterprise license wasn't needed for multi instance?

**Comment by mickmister** [comment]
@aaronrothschild I've changed the docs so it's clear that E20 _was_ required and now only an Enterprise license is required.

**Review by aaronrothschild — APPROVED** [review]
LGTM

File: server/enterprise/license.go [hunk 2]
```diff
@@ -24,8 +24,19 @@ func NewEnterpriseChecker(api PluginAPI) Checker {
...
 func (e *enterpriseChecker) HasEnterpriseFeatures() bool {
 	config := e.api.GetConfig()
 	license := e.api.GetLicense()
-	return pluginapi.IsE10LicensedOrDevelopment(config, license)
+
+	if license != nil && (license.SkuShortName == e20 || license.SkuShortName == enterprise || license.SkuShortName == professional) {
+		return true
+	}
+
+	return pluginapi.IsE20LicensedOrDevelopment(config, license)
```
Comment (marianunez) [inline server/enterprise/license.go:41]: Will this call make any different checks than the above?

Comment (mickmister) [reply server/enterprise/license.go:41]: @marianunez Yes, see the code at https://github.com/mattermost/mattermost-plugin-api/blob/1aa5f947aaa680fdfcf8bfae20b2bbe17b69517e/license.go#L67-L98. It is checking for E20 features in the case that the license doesn't have a SKU present. It also checks if the system is in development mode as the final check, if the license isn't present or isn't the kind of license we're looking for.

Comment (marianunez) [reply server/enterprise/license.go:41]: Nice, thanks for clarifying!

**Comment by marianunez** [comment]
@DHaussermann I can provide you licenses for the new skus if you need for testing this one - just let me know

**Comment by DHaussermann** [comment]
Thank @marianunez that would be much appreciated. Maybe you could DM the license files?

**Review by aaronrothschild — APPROVED** [review]

**Review by DHaussermann — APPROVED** [review]
Tested and passed 
All 9 licenses are working as expected.

- Tested 3 legacy license types (unchanged)
  - E20 Allows multiple Jira instance installs
  - E10 Does **not** allow multiple Jira instance installs
   - E0 Does **not** allow multiple Jira instance installs
- Tested  both new Cloud and On-Prem SKU sets. For both: 
  - Enterprise Allows multiple Jira instance installs
  - Professional Allows multiple Jira instance installs
  - Starter Does **not** allow multiple Jira instance installs


LGTM!
Please merge

**Pull request merged**